File: Ocena 5 Marciniak Daniel/zad1.ts
function sumOfSquares(n: number): number {
    return Array.from({ length: n }, (_, i) => (i + 1) ** 2).reduce((a, b) => a + b, 0);
}

function squareOfSum(n: number): number {
    return Math.pow(Array.from({ length: n }, (_, i) => i + 1).reduce((a, b) => a + b, 0), 2)
}

function difference(n: number): number {
    return squareOfSum(n) - sumOfSquares(n);
}

const n = 100; // it is const
console.log(`Difference: ${difference(n)}`); // can be diffrence(100), then const n can be remove

File: Ocena 3.5 Kasprzak Sebastian/Zad1.java
package prod.func;

public class Zad1 {
    public static void main(String[] args) 
    {
        System.out.println("Difference: " + (squareOfSum(100) - sumOfSquares(100)));
    }

    public static int sumOfSquares(int n) {
        return (n == 1) ? 1 : n * n + sumOfSquares(n - 1);
    }

    public static int squareOfSum(int n) {
        return sum(n) * sum(n);
    }

    public static int sum(int n) {
        return (n == 1) ? 1 : n + sum(n - 1);
    }
}
